### ISSUE TITLE
workflow

### ISSUE THREAD

**Issue by rafinskipg** [bq/composr-core#3]
Now that we have some basic bricks built, lets think about the workflow:
- User enters credentials and initializes composr-core
  - composr checks if the user is allowed to get phrases.
  - _if yes_ : notifies a succesfull login and continues
  - _if not_ : notifies invalid credentials
- User fetches phrases
  - composr checks if the user is allowed to access multiple domains,
    - _if yes_: loads all the phrases for multiple domains, stablishes multidomain = true
    - _if not_ : loads the phrases for that current domain, stablishes multidomain = false and the default domain.
- User publishes phrases
  - composr checks if the user is allowed to publish
    - _if yes_ : publishes phrases and notifies the number of phrases published by domain
      -_if not_ : notifies that the permissions for publishing are wrong
- User deletes a phrase:
  - composr checks if the user is allowed to delete
    - _if yes_ : deletes phrases and notifies the number of phrases deleted by domain
      -_if not_ : notifies that the permissions for delleting are wrong
- User request in memory phrases: 
  - User is `multipledomain === true`
    - _yes_ : user can access phrases from all the domains
    - _no_ : there is only one domain loaded, it can only request the phrases for that domain

Let's notify mr anthanh about this and wait for his opinion.
@anthanh

**Comment by anthanh** [comment]
Looks great! @rafinskipg, but I don't really understand the meaning of "multiple-domains", maybe it could be achieved with the right credentials (client+secret+scopes)? one scope with full access, and other one with domain specific...

Maybe we should identify different user-roles where composr-core works, to understand use cases better. (composr-server, composr-phrases-test, app-that-execute-phrases, ...).

Rest of the issue looks fine, we will continue working on it.

**Comment by rafinskipg** [comment]
Yeah, multiple domains means that the credentials are allowed to access to all the domains, and the other is domain specific.

Setting this flag is just for having a default domain to work with in the case you dont have permissions to perform operations on multiple domains.

Cases:
- Composr-server : 
  - multiple domains actions
- composr-cli  (and therefore ,grunt-composr... etc)
  - depending on your credentials, but mostly single domain actions.

**Comment by rafinskipg** [comment]
As talked 
- Domain is always mandatory
  - composr-cli can handle currying the methods with the default domain
- Changed method run for:
  - runById
  - runByPath
